### PR TITLE
fixes #4568 feat(nimbus): Update mutation probe set IDs to slugs

### LINKED ISSUE #4568: (BE & FE) Update mutation from ID to slug
https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/api/v5/inputs.py#L50-L51
https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/PageEditMetrics/FormMetrics/index.tsx#L74
https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/api/v5/serializers.py#L177-L231

We need to change the references of probe set IDs to probe set slugs in the BE & FE since slugs (and not IDs) will be pulled from the outcome data.

Note updating the read references will be taken care of in EXP-925.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-923)
┆Issue Number: EXP-923

### PR DESCRIPTION
**Pull request by LZoog**
Because:
* Slugs, not IDs, will be pulled from the outcomes data when probe sets are replaced with outcomes

This commit:
* Updates references to probe sets IDs to slugs in relevant mutation and corresponding files, FE & BE

fixes #4568

---

~Locally I'm getting a complaint that `PageFormMetrics` doesn't meet the coverage threshold which seems strange but ~opening as a draft to get @jaredlockhart's eyes in case I'm missing something here (this seemed a little more straightforward than I figured it'd be).~

### CODE REVIEW

File: app/experimenter/experiments/api/v5/serializers.py [hunk 1]
```diff
@@ -305,12 +305,19 @@ class NimbusExperimentSerializer(
...
         required=False,
     )
-    primary_probe_set_ids = serializers.PrimaryKeyRelatedField(
-        many=True, queryset=NimbusProbeSet.objects.all(), required=False
+
+    primary_probe_set_slugs = serializers.SlugRelatedField(
+        many=True,
+        queryset=NimbusProbeSet.objects.all(),
+        required=False,
+        slug_field="slug",
```
Comment (jaredlockhart) [inline app/experimenter/experiments/api/v5/serializers.py:313]: Great job 👍

File: app/experimenter/experiments/tests/api/v5/test_mutations.py [hunk 2]
```diff
@@ -522,8 +522,8 @@ def test_update_experiment_probe_sets_error(self):
             variables={
                 "input": {
                     "id": experiment.id,
-                    "primaryProbeSetIds": [123],
-                    "secondaryProbeSetIds": [],
+                    "primaryProbeSetSlugs": ["my-lovely-slug"],
```
Comment (jaredlockhart) [inline app/experimenter/experiments/tests/api/v5/test_mutations.py:525]: image

File: app/experimenter/nimbus-ui/src/components/PageEditMetrics/index.tsx [hunk 3]
```diff
@@ -28,23 +28,23 @@ const PageEditMetrics: React.FunctionComponent<RouteComponentProps> = () => {
...
-              secondaryProbeSetIds,
+              primaryProbeSetSlugs,
+              secondaryProbeSetSlugs,
             },
           },
         });
 
         if (!result.data?.updateExperiment) {
-          throw new Error("Save failed, no error available");
+          throw new Error(SUBMIT_ERROR);
```
Comment (jaredlockhart) [inline app/experimenter/nimbus-ui/src/components/PageEditMetrics/index.tsx:47]: Good catch 👍

**Review by jaredlockhart — APPROVED** [review]
All looks right to me, great job on using SlugRelatedField on the serializer that's exactly right ;)  Nice that it was almost entirely substitutions.

File: app/experimenter/nimbus-ui/src/components/PageEditMetrics/index.tsx (hunk 3 above)
Comment (pdehaan) [reply app/experimenter/nimbus-ui/src/components/PageEditMetrics/index.tsx:47]: Now you done got me curious... Here's the other places where we seemingly throw new Errors (some of which with hardcoded strings, some not; potentially none relevant):

```sh
git grep -n "throw new " app/experimenter/nimbus-ui

app/experimenter/nimbus-ui/README.md:158:    throw new Error("Update failed for an unknown reason");
app/experimenter/nimbus-ui/src/components/AppErrorBoundary/index.test.tsx:42:      throw new Error("bad");
app/experimenter/nimbus-ui/src/components/PageEditAudience/index.tsx:57:          throw new Error(SUBMIT_ERROR);
app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/index.test.tsx:110:      throw new Error(expectedErrorMessage);
app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/reducer/update.ts:27:    throw new UpdateStateError("Control branch is required");
app/experimenter/nimbus-ui/src/components/PageEditBranches/index.tsx:58:          throw new Error(SUBMIT_ERROR_MESSAGE);
app/experimenter/nimbus-ui/src/components/PageEditMetrics/index.tsx:47:          throw new Error("Save failed, no error available");
app/experimenter/nimbus-ui/src/components/PageEditOverview/index.tsx:53:          throw new Error("Save failed, no error available");
app/experimenter/nimbus-ui/src/components/PageNew/index.tsx:43:          throw new Error("Save failed, no error available");
app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:97:        throw new Error(SUBMIT_ERROR);
app/experimenter/nimbus-ui/src/index.test.tsx:71:      throw new Error("uh oh");
app/experimenter/nimbus-ui/src/services/config.ts:30:      throw new Error("Configuration is empty");
app/experimenter/nimbus-ui/src/services/config.ts:42:      throw new Error(`Invalid configuration ${JSON.stringify(content)}: ${decoded}`,
```

Comment (LZoog) [reply app/experimenter/nimbus-ui/src/components/PageEditMetrics/index.tsx:47]: @pdehaan I noticed that too. We have this string hanging out in a few places as well as called `SUBMIT_ERROR_MESSAGE` in another file. I'll create a follow up (good maintenance item!) to move them in our `constants` file. :)

Comment (LZoog) [reply app/experimenter/nimbus-ui/src/components/PageEditMetrics/index.tsx:47]: Issue here: #4597

**Pull request merged**